cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/MetricsHandlers.java

```diff
@@ -0,0 +1,213 @@
+        
+        // Get the reset parameter
+        String reset = request.getParameter("reset");
+        
+        if (reset != null && !reset.equals("true") && !reset.equals("false")) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        } // if
+        
+        if (reset != null && reset.equals("true")) {
+            CygnusMetrics metrics = mergeMetrics(sources, sinks);
+            response.getWriter().println(metrics.toJsonString());
+            deleteMetrics(sources, sinks);
+            response.setContentType("application/json; charset=utf-8");
+        } else {
+            CygnusMetrics metrics = mergeMetrics(sources, sinks);
+            response.getWriter().println(metrics.toJsonString());
+            response.setContentType("application/json; charset=utf-8");
+        } // else
+    } // getServiceSubserviceMetrics
+    
+    /**
+     * Handles DELETE /v1/admin/mergeMetrics and /admin/mergeMetrics.
+     * @param response
+     * @param sources
+     * @param sinks
+     * @throws java.io.IOException
+     */
+    public static synchronized void delete(HttpServletResponse response, ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) throws IOException {
+        deleteMetrics(sources, sinks);
+        response.setStatus(HttpServletResponse.SC_OK);
+    } // delete
+    
+    /**
+     * Merges metrics from all the given sources and sinks. It is protected in order it can be tested.
+     * @param sources
+     * @param sinks
+     * @return
+     */
+    protected static CygnusMetrics mergeMetrics(ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) {
+        CygnusMetrics mergeServices = new CygnusMetrics();
+    
+        if (sources != null) {
+            for (String key : sources.keySet()) {
+                Source source;
+                HTTPSourceHandler handler;
+
+                try {
+                    SourceRunner sr = sources.get(key);
+                    source = sr.getSource();
+                    Field f = source.getClass().getDeclaredField("handler");
+                    f.setAccessible(true);
+                    handler = (HTTPSourceHandler) f.get(source);
+                } catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException
+                        | SecurityException e) {
+                    LOGGER.error("There was a problem when getting a sink. Details: " + e.getMessage());
+                    continue;
+                } // try catch
+
+                if (handler instanceof CygnusHandler) {
+                    CygnusHandler ch = (CygnusHandler) handler;
+                    CygnusMetrics sourceServices = ch.getServiceMetrics();
+                    mergeServices.merge(sourceServices);
+                } // if
+            } // for
+        } // if
+        
+        if (sinks != null) {
```

Comment:
Yes, you are correct. Sources gather information about the number of incoming transactions and  request and response sizes. While sinks gather information about the service time. Errors are handled by both components, since errors may be detected at the moment of receiving a notification (i.e. the target in the URL is not correct, or the content-type is invalid) or at the moment of doing the persistence (the backend is not available, or there is a network error).